selfservice/strategy/oidc/provider.go

```diff
@@ -65,3 +65,14 @@ func (c *Claims) Validate() error {
 	}
 	return nil
 }
+
+func UpstreamParameters(provider Provider, upstreamParameters map[string]string) []oauth2.AuthCodeOption {
+	// validation of sent providers are already handled in the `oidc/.schema/link.schema.json` file.
+	// `upstreamParameters` will always only contain allowed parameters based on the configuration.
```

Comment:
I'd prefer this to be more explicit, because the input source probably is validated against the form payload, but it might not be in a certain case (e.g. the flow is resumed or something). By putting the validation in **the** central place where the code is executed, the input source no longer matters because the validation is "deeply embedded" into the flow, and not on the "surface".

The tests cover this nicely already, but for example they don't cover the account settings flow. So here we would have uncertainty whether this validation actually is implemented / working.

Generally speaking, **security validation** should always be very very explicit and treated like "do you actually have this permission and do I have 100% certainty?".

Reply:
I believe this should now be addressed. I have added the upstream parameters to Settings flow as well. I am also double checking the parameters inside the `UpstreamParameters` function.